### PR TITLE
Replace JFrog artifactory auth from basic to token-based

### PR DESCRIPTION
**Pull request by andrei-epure-sonarsource**

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -119,11 +122,11 @@ stages:
           pathToPublish: '$(Build.ArtifactStagingDirectory)/packages'
           artifactName: 'NuGet Packages'
 
-      - task: ArtifactoryNuGet@1
+      - task: JFrogNuGet@1
         displayName: 'Publish SonarAnalyzer.CFG NuGet on repox'
         inputs:
           command: 'push'
-          artifactoryService: 'Repox artifactory'
+          artifactoryConnection: 'repox_qa_deployer_token'
```
Comment (andrei-epure-sonarsource) [inline azure-pipelines.yml:129]: @malena-ebert-sonarsource  here I configured the artifactory connection, but for Maven deploy, I didn't change it...

I am not sure it uses the new connection?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -26,6 +27,8 @@ variables:
     value: ${{ or(eq(variables['Build.SourceBranch'], 'refs/heads/master'), startsWith(variables['Build.SourceBranch'], 'refs/heads/branch-')) }}
   - name: vsVersion
     value: '17.0'
+  - name: ARTIFACTORY_DEPLOY_USERNAME
```
Comment (andrei-epure-sonarsource) [inline azure-pipelines.yml:30]: we use this to overwrite the setting from the first user group

**Comment by andrei-epure-sonarsource** [comment]
Also, after this PR gets merged, I will need to delete the old usernames from the old variable group:

Temp: use the old names, and after we do the switch, we delete the old variables and use the new ones.

ARTIFACTORY_DEPLOY_USERNAME
ARTIFACTORY_API_USER
ARTIFACTORY_PRIVATE_READER_USERNAME
ARTIFACTORY_QA_READER_USERNAME

Making sure I don't break anything.

File: azure-pipelines.yml [hunk 3]
```diff
@@ -482,11 +491,11 @@ stages:
 
       - template: set-azp-variables-steps.yml@pipelines-yaml-templates
 
-      - task: Maven@3
+      - task: JFrogMaven@1
```
Comment (andrei-epure-sonarsource) [inline azure-pipelines.yml:494]: @elrzn I am not sure we need the deployment connection for this... I will do a test on a separate PR.

But if it doesn't hurt, I suggest to merge it as is and make changes later.

File: azure-pipelines.yml [hunk 4]
```diff
@@ -401,6 +404,12 @@ stages:
           javaHomeOption: 'JDKVersion'
           jdkVersionOption: '1.17'
           mavenOptions: $(MAVEN_OPTS)
+          artifactoryDeployService: 'repox_qa_deployer_token'
```
Comment (andrei-epure-sonarsource) [inline azure-pipelines.yml:407]: I added these arguments to please the tool, I hope it won't hurt

File: azure-pipelines.yml [hunk 5]
```diff
@@ -482,11 +491,11 @@ stages:
 
       - template: set-azp-variables-steps.yml@pipelines-yaml-templates
 
-      - task: Maven@3
+      - task: JFrogMaven@1
         displayName: 'Maven install'
         env:
           ARTIFACTORY_DEPLOY_USERNAME: $(ARTIFACTORY_DEPLOY_USERNAME)
-          ARTIFACTORY_DEPLOY_PASSWORD: $(ARTIFACTORY_DEPLOY_PASSWORD)
+          ARTIFACTORY_DEPLOY_PASSWORD: $(ARTIFACTORY_PRIVATE_READER_ACCESS_TOKEN)
```
Comment (andrei-epure-sonarsource) [inline azure-pipelines.yml:498]: @elrzn TBH here I don't understand why the DEPLOY username works with the PRIVATE_READER token

Comment (elrzn) [reply azure-pipelines.yml:498]: If you are using `artifactoryDeployService`, try removing `ARTIFACTORY_DEPLOY_USERNAME` and `ARTIFACTORY_DEPLOY_PASSWORD` as they seem to be mutually exclusive.

Comment (andrei-epure-sonarsource) [reply azure-pipelines.yml]: I removed them and it fails

> [main] ERROR org.apache.maven.plugins.enforcer.EnforceMojo - Rule 4: org.apache.maven.plugins.enforcer.RequireEnvironmentVariable failed with message:
Environment variable "ARTIFACTORY_DEPLOY_PASSWORD" is required for this build.

https://dev.azure.com/sonarsource/DotNetTeam%20Project/_build/results?buildId=76559&view=logs&jobId=5a84c39f-87b6-5ab2-d0a7-dae390472f36&j=5a84c39f-87b6-5ab2-d0a7-dae390472f36&t=751e0046-590d-5d14-2e2f-85058296ad91

**Comment by andrei-epure-sonarsource** [comment]
fixed in https://github.com/SonarSource/sonar-dotnet/pull/7919

**Pull request closed without merging**